### PR TITLE
feat: support send and recv msg size for gRPC

### PR DESCRIPTION
**Pull request by alexanderbez**
## Description

Update the gRPC configuration to support providing user-provided receive and send max message sizes.

Note, this should be upstreamed to the Cosmos SDK core repo.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: server/config/toml.go [hunk 1]
```diff
@@ -176,6 +176,14 @@ enable = {{ .GRPC.Enable }}
 # Address defines the gRPC server address to bind to.
 address = "{{ .GRPC.Address }}"
 
+# MaxRecvMsgSize defines the max message size in bytes the server can receive.
+# The default value is 4MB.
+max-recv-msg-size = "{{ .GRPC.MaxRecvMsgSize }}"
+
+# MaxSendMsgSize defines the max message size in bytes the server can send.
+# The default value is math.MaxInt32.
```
Comment (p0mvn) [inline server/config/toml.go:184]: asking to learn - what is the reason behind the choice of this default for the send and only 4MB for the receive?

Comment (p0mvn) [inline server/config/toml.go:184]: asking to learn - what is the reason behind the choice of this default for the send and only 4MB for the receive?

Comment (alexanderbez) [reply server/config/toml.go:184]: It's actually the gRPC default. In fact, the entire comment is more or less copied from there.

Ref: https://github.com/grpc/grpc-go/blob/a73725f42db97964eb538a5cf4d302f760357dbe/server.go#L58-L59

**Review by ValarDragon — APPROVED** [review]

File: server/config/config.go [hunk 2]
```diff
@@ -224,8 +233,10 @@ func DefaultConfig() *Config {
 			RPCMaxBodyBytes:    1000000,
 		},
 		GRPC: GRPCConfig{
-			Enable:  true,
-			Address: DefaultGRPCAddress,
+			Enable:         true,
+			Address:        DefaultGRPCAddress,
+			MaxRecvMsgSize: 1024 * 1024 * 4,
```
Comment (ValarDragon) [inline server/config/config.go]: Should we increase this to 10MB in light of our epoch time issues?

**Comment by alexanderbez** [comment]
I have no idea what's up with `test-rosetta`. I run it locally, and there are absolutely zero logs. There is also no _current_ team maintaining it. I suggest we just remove it all together. Thoughts @ValarDragon?

**Pull request merged**